sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -377,8 +377,13 @@ case class AlterTableChangeColumnCommand(
     val resolver = sparkSession.sessionState.conf.resolver
     DDLUtils.verifyAlterTableType(catalog, table, isView = false)
 
-    // Find the origin column from dataSchema by column name.
+    // check that the column is not a partition column
+    if (findColumnByName(table.partitionSchema, columnName, resolver).isDefined) {
```

Comment:
How about `table.partitionSchema.fieldNames.exists(resolver(_, columnName))` ?

So we no need change `findColumnByName` and related code.